DeliBuddy/app/src/main/java/yapp/android1/delibuddy/DeliBuddyApplication.kt

```diff
@@ -6,11 +6,16 @@ import com.naver.maps.map.NaverMapSdk
 import dagger.hilt.android.HiltAndroidApp
 import timber.log.Timber
 import timber.log.Timber.Forest.plant
+import yapp.android1.delibuddy.util.sharedpreferences.SharedPreferencesManager
 
 @HiltAndroidApp
 internal class DeliBuddyApplication : Application() {
+    companion object {
+        lateinit var prefs: SharedPreferencesManager
```

Comment:
```suggestion
        val prefs: SharedPreferencesManager = SharedPreferencesManager(applicationContext)
```
이렇게 하는건 어떨까요?

Reply:
밖에서도 applicationContext를 받아올 수 있나요?

Reply:
혹시 안되셔서 저렇게 하셨던건가요?? 저도 지금 아리까리 해서 ㅋㅋㅋ

Reply:
네 ㅋㅋ쿠ㅜㅜ onCreate 안에서만 컨텍스트를 받아오는 게 가능하더라고요 😞

Reply:
https://stackoverflow.com/questions/54075649/access-application-context-in-companion-object-in-kotlin) 여기 첫번째 답변처럼 하는게 좀 더 깔끔할거 같네요!

Reply:
혹시 onCreate()에서 prefs 관련 부분을 지우고 companion object 부분만 수정하면 되는 게 맞을까요?
제일 첫 번째 답변처럼 수정해 보았는데 prefs를 초기화해 주는 부분에서 null pointer exception이 발생하는 것 같습니다

+ 개인적인 궁금함인데 lateinit으로 초기화하는 것이 좋지 않은 습관인지..?
문제가 발생할 수 있는 여지를 주는 것인지 궁금합니당
영상 님 코드를 봤을 때 lateinit을 거의 사용하지 않으시고 보통 nullable type으로 지정 후 null로 초기화하여 사용하시는 것 같아서 보면서 궁금했었던 부분이었어요!

+ 첨부 파일은 제가 영상 님이 공유해 주신 글을 보고 수정해 본 코드입니다(여기서 오류가 발생해요 ㅠㅠ)
혹시 제가 잘못 이해하고 수정한 것일까 해서 공유해 봅니다.. ㅠㅠ
<img width="764" alt="스크린샷 2021-12-02 오후 7 22 00" src="https://user-images.githubusercontent.com/42907876/144403558-17554506-696c-4da9-9445-f956be007b2d.png">

Reply:
- lateinit이 잘못된게 절대 아니고 제가 짜는 부분에 Base부분이라 Nullable이 필요한곳엔 꼭 쓰는거에요!
   ex) Fragment 에서는 binding 을 onDestroyView에서 null로 만들어 줘야 할때 
   Null을 꼭 필요로 하는 부분들이 있다고 생각하는데 이런부분들 에서는 Nullable을 사용하는 편입니다.
   lateinit은 nullable 타입을 허용하지 않기 때문에 제가 Nullbale을 많이 사용했는데
   이 부분 때문에 lateinit을 안쓰는것처럼 보였겠네요!
   
   저도 adapter나 이런것들은  lateinit을 사용하고 합니다!
   
- 이 부분 다시 생각해보니 제가 전혀 다른 부분을 말씀드리고 있었네요
   죄송합니다.. 원복 하시면 될것 같아요..
   
   다시 하나 제안을 드리자면 
```suggestion
        val prefs: SharedPreferencesManager by lazy { SharedPreferencesManager(applicationContext) }
```
로 한번 해보시면 어떨까요..?
Application의  onCreate에서  super.onCreate()이전에 생성하길래 뭔가 좀 불안정 해 보이고
덜 깔끔해 보여서 위에 방법 제안을 드려봅니다.

Reply:
아하 그랬군요..! 제가 영상 님 코드를 보면서 잘못 이해한 것 같네요!
아 super.onCreate() 이전에 생성하는 부분은 저도 아까 수정하면서 발견했습니다.. ㅋㅋ큐ㅠ
super.onCreate()과 prefs 초기화해 주는 부분의 순서를 바꿔야 할 것 같아요!

Reply:
위에 Suggestion처럼 한번 해보시고 안되시면 조용히 물러나겠습니다..

Reply:
앗 다시 제안해 주신 부분은 아까 처음 말씀드린 대로 `applicationContext`를 밖에서 사용할 수가 없어서 안 되는 것 같아요..!
그런데 처음에 이 리뷰가 달렸던 이유가 onCreate에서
`prefs = SharedPreferencesManager(applicationContext)` 이 부분이 
`super.onCreate()` 이전에 하도록 되어 있어서 그랬던 것 같아요
이거는 제가 실수로 순서를 잘못 작성한 것 같은데
`super.onCreate()` 이후에 `prefs = SharedPreferencesManager(applicationContext)`를 하도록 수정했습니다!

Reply:
에고.. 괜히 말씀드려서 제가 귀찮게 했네요!
알겠습니다! 만약에 더 좋은 방법이 있다면 그때 리팩토링 하면 좋을것 같아요
오늘도 하나 배워갑니다..

Reply:
아니에요..! 덕분에 순서 잘못되어 있었던 부분을 발견했습니다!! 👍
좋아유 나중에 더 좋은 방법이 떠오른다면 다시 수정해 봅시당 😄